### PR TITLE
Implement Pagination and search

### PR DESCRIPTION
**Pull request by TuyizeeAnastase**
### What does this PR do?

- Users and car parking pagination and searching

### Description of tasks to be completed

- Admin must get all users and car parkings using pagination
- Admin could search user and car parking 

### How can be manually tested

-use postman:
- get all users and searc: localhost:3500/api/v1/users/clients?limit=4&page=0&q=c
- get all car parking and search:     localhost:3500/api/v1/parking?page=0&limit=1&q=RAB

### What are trello story?
https://trello.com/c/IbOlDt9f/28-implement-parking

### CODE REVIEW

File: src/controllers/parking.controller.js [hunk 1]
```diff
@@ -2,13 +2,30 @@ import {
...
+      }
+      else{
+        response = await getParkings(limit,offset);
+      }
+      const parkings = getPagingData (response,page,limit)
+      if(parkings.totalItems <= 0){
+        return res.status(203).json({
+          message:'Search results not found'
+        })
+      }
```
Comment (ncutixavier) [inline src/controllers/parking.controller.js:28]: I think, this is not necessary

File: src/controllers/user.controller.js [hunk 2]
```diff
@@ -145,6 +148,37 @@ export class userControllers {
...
+      }
+      else{
+        response = await getAllUsers(limit,offset)
+      }
+      const users = getPagingData(response,page,limit)
+      if(users.totalItems <= 0){
+        return res.status(203).json({
+          message:'Search results not found'
+        })
+      }
```
Comment (ncutixavier) [inline src/controllers/user.controller.js:169]: Why do we need this?

**Review by ncutixavier — CHANGES_REQUESTED** [review]

File: src/utils/pagination.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+export const getPagination = (page,size)=>{
+    const limit = size ? +size:1
+    const offset = page ? page * limit:0;
+    
+    return { limit,offset }
+}
+
+export const getPagingData = (data, page, limit) => {
+    const { count: totalItems, rows: response } = data;
+    const currentPage = page ? +page : 0;
```
Comment (ncutixavier) [inline src/utils/pagination.js:11]: Default page number should be 1.

`localhost:3500/api/v1/parking?page=0&size=1&q=RAC` : page should be 1

**Review by ncutixavier — CHANGES_REQUESTED** [review]

File: src/utils/pagination.js (hunk 3 above)
Comment (ncutixavier) [reply src/utils/pagination.js:11]: Can you rename `size` to `limit`

Comment (ncutixavier) [reply src/utils/pagination.js:11]: Your query `q` should be case sensitive

Comment (ncutixavier) [reply src/utils/pagination.js:11]: Also I can't see get all clients

Comment (TuyizeeAnastase) [reply src/utils/pagination.js]: > Also I can't see get all clients

localhost:3500/api/v1/users/clients

Comment (TuyizeeAnastase) [reply src/utils/pagination.js]: localhost:3500/api/v1/users/clients?limit=4&page=0&q=c

**Review by ncutixavier — APPROVED** [review]

**Pull request merged**